### PR TITLE
move vote counting out of rate limiting zone

### PR DESCRIPTION
**Pull request by longbowlu**
vote recording and finality check has no network impact and should not be rate limited

### CODE REVIEW

File: crates/sui-core/src/node_sync/node_state.rs [hunk 1]
```diff
@@ -370,46 +425,55 @@ where
...
 
+    fn send_sync_res_to_receiver(
+        tx: Option<oneshot::Sender<Result<SyncStatus, SuiError>>>,
+        res: Result<SyncStatus, SuiError>,
+        sync_arg: &SyncArg,
+        epoch_id: &EpochId,
+    ) {
+        if let Some(tx) = tx {
+            // Send status back to follower so that it knows whether to advance
+            // the watermark.
```
Comment (mystenmark) [inline crates/sui-core/src/node_sync/node_state.rs:463]: fwiw this comment is obsolete, there's no water mark anymore. I'm not sure any comment is necessary here at all.

**Review by mystenmark — APPROVED** [review]
FYI I usually test changes to this code by patching my change into the devnet branch, building a full node, and verifying that:
- It can sync from scratch 3 times in a row without getting stuck
- the throughput is ok (instructions for checking throughput here: https://github.com/MystenLabs/sui-operations/pull/193/files)
- It can resume from an interrupted sync.

**Comment by longbowlu** [comment]
tested last night and all is good. The performance boost is not as big as I imagine. Before this change, the average transactions synced per second, by eyeballing is around ~80. With this change, it's 100-110.

I also tested spawning a tokio task for vote counting but only requires the (downloading) permit when it's final. The idea is trying to see whether counting votes in the single thread is blocking. Assuming my code is correct, the results are weird - the throughput is fairly unstable, in some seconds it has a couple of thousands but in most the time, it's like single digits. My conjecture is, if we unboundedly spawn tokio tasks (especially when the qps is high), the latter transactions could be finalized before their ancestors and put to suspension, causing big unstableness in throughput.

**Comment by longbowlu** [comment]
will be doing another test in staging with 100 qps

**Pull request merged**